Source/WebCore/html/OffscreenCanvas.cpp

```diff
@@ -232,11 +232,11 @@ ExceptionOr<std::optional<OffscreenRenderingContext>> OffscreenCanvas::getContex
             Ref scriptExecutionContext = *this->scriptExecutionContext();
             if (RefPtr globalScope = dynamicDowncast<WorkerGlobalScope>(scriptExecutionContext)) {
                 if (auto* gpu = globalScope->protectedNavigator()->gpu())
-                    m_context = GPUCanvasContext::create(*this, *gpu);
```

Comment:
Poor workers :(

Worth filing a bug with a fixme, to get worker contexts notified of the change?

Reply:
I am wondering about this, isn't the worker displayed in a Document via an OffscreenCanvas? But I was not sure how to obtain the Document the OffscreenCanvas is displayed in, that was not immediately obvious to me

Reply:
It’s usually in a separate thread, so we cant access the Document. We’d need a threadsafe way to notify these observers

Reply:
Let me see if this is easily resolvable as it would be good to have this behavior for workers too